### PR TITLE
Update compilation instructions

### PR DESCRIPTION
**Pull request by christophfroehlich**

### CODE REVIEW

**Review by ahcorde — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -37,12 +37,12 @@ Note that `gz_ros2_control` depends on the version of Gazebo that is
...
 Currently, for ROS 2 Jazzy and Rolling, the Gazebo version is Harmonic.
 
-To compile `gz_ros2_control` from source, create a workspace, clone the repo and compile it:
+To compile `gz_ros2_control` from source, create a workspace, clone the correct branch of this repo and compile it:
 
 ```bash
 mkdir -p ~/gz_ros2_control_ws/src
 cd ~/gz_ros2_control_ws/src
-git clone https://github.com/ros-controls/gz_ros2_control
+git clone https://github.com/ros-controls/gz_ros2_control -b jazzy
```
Comment (ahcorde) [inline README.md:45]: what about 
```suggestion
git clone https://github.com/ros-controls/gz_ros2_control -b {ROS_DISTRO}
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (ahcorde) [reply README.md:45]: this change is backportable

Comment (christophfroehlich) [reply README.md]: yes that is true, but it doesn't work on master so I thought I can hardcode that as well

**Review by ahcorde — APPROVED** [review]

**Comment by ahcorde** [comment]
https://github.com/Mergifyio backport humble iron

**Pull request merged**